hbase-client/src/main/java/org/apache/omid/transaction/TTable.java

```diff
@@ -321,16 +317,51 @@ public void put(Transaction tx, Put put) throws IOException {
                 Bytes.putLong(kv.getValueArray(), kv.getTimestampOffset(), writeTimestamp);
                 tsput.add(kv);
 
-                transaction.addWriteSetElement(
+                updateMetaData.update(transaction, kv, tsput);
+            }
+        }
+
+        table.put(tsput);
+    }
+
+    /**
+     * putWithAutocommit implementation. Similar to a transactional put that includes the commit time stamp.
+     *
+     * @param put an instance of Put
+     * @param tx  an instance of transaction to be used
+     * @throws IOException if a remote or network exception occurs.
+     */
+    public void putWithAutocommit(Transaction tx, Put put) throws IOException {
```

Comment:
Not sure it's a good idea to have a separate method for this. Is it painful to add a parameter? Alternatively, maybe you could pass an attribute to propagate (mechanism defined in OMID-83)?